support/src/main/java/org/jenkinsci/plugins/workflow/support/steps/StageStepExecution.java

```diff
@@ -147,6 +143,20 @@ private static synchronized void enter(Run<?,?> r, StepContext context, String n
                 throw new IllegalStateException("the same flow is trying to reënter the stage " + name); // see 'e' with two dots, that's Jesse Glick for you! - KK
             }
         }
+        // Interrupts any build in the holding state
+        if (eager != null && eager) {
+            for (Integer i : stage.holding) {
+                try {
+                    Executor ex = job.getBuildByNumber(i).getExecutor();
+                    if (ex != null) {
+                        ex.interrupt(Result.ABORTED);
+                    }
+                    stage.holding.remove(i);
```

Comment:
Need to use `Iterator.remove` to avoid `ConcurrentModificationException` I think.